Libplanet/Net/Messages/RecentStates.cs

```diff
@@ -0,0 +1,223 @@
+                    refs.Add(new HashDigest<SHA256>(it.Current.Buffer));
+                }
+
+                stateRefs[address] = refs.ToImmutableList();
+            }
+
+            it.MoveNext();
+            int signersCount = BitConverter.ToInt32(it.Current.Buffer, 0);
+
+            var txNonces = new Dictionary<Address, long>(accountsCount);
+
+            for (int j = 0; j < signersCount; j++)
+            {
+                it.MoveNext();
+                var address = new Address(it.Current.Buffer);
+
+                it.MoveNext();
+                txNonces[address] = BitConverter.ToInt64(it.Current.Buffer, 0);
+            }
+
+            it.MoveNext();
+            int blocksLength = BitConverter.ToInt32(it.Current.Buffer, 0);  // This is not height!
+
+            var blockStates =
+                new Dictionary<HashDigest<SHA256>, IImmutableDictionary<Address, object>>(
+                    blocksLength);
+            var formatter = new BinaryFormatter();
+
+            for (int j = 0; j < blocksLength; j++)
+            {
+                it.MoveNext();
+                var blockHash = new HashDigest<SHA256>(it.Current.Buffer);
+
+                it.MoveNext();
+                int statesLength = BitConverter.ToInt32(it.Current.Buffer, 0);
+
+                var states = new Dictionary<Address, object>(statesLength);
+                for (int k = 0; k < statesLength; k++)
+                {
+                    it.MoveNext();
+                    var address = new Address(it.Current.Buffer);
+
+                    it.MoveNext();
+                    using (var stream = new MemoryStream(it.Current.Buffer))
+                    {
+                        states[address] = formatter.Deserialize(stream);
+                    }
+                }
+
+                blockStates[blockHash] = states.ToImmutableDictionary();
+            }
+
+            BlockStates = blockStates.ToImmutableDictionary();
+            StateReferences = stateRefs.ToImmutableDictionary();
+            TxNonces = txNonces.ToImmutableDictionary();
+        }
+
+        public HashDigest<SHA256> BlockHash { get; }
+
+        public bool Missing => BlockStates is null;
+
+        public IImmutableDictionary<
+            HashDigest<SHA256>,
+            IImmutableDictionary<Address, object>
+        > BlockStates { get; }
+
+        /// <summary>
+        /// State references of all available accounts.  Each value has a list of
+        /// state references in ascending order; the closest to the genesis block
+        /// goes first, and the closest to the tip goes last.
+        /// </summary>
+        public IImmutableDictionary<Address, IImmutableList<HashDigest<SHA256>>> StateReferences
+        {
+            get;
+        }
+
+        public IImmutableDictionary<Address, long> TxNonces { get; }
+
+        protected override MessageType Type => MessageType.RecentStates;
+
+        protected override IEnumerable<NetMQFrame> DataFrames
```

Comment:
The payload structure is quite complex. How about writing down a brief description of the structure?

Reply:
Wrote! 4f2cdb9e3aa081c1e9377062e1dba4f17aebb0a8